pkg/aws/actuator/actuator.go

```diff
@@ -263,11 +264,82 @@ func (a *AWSActuator) getProviderSpec(cr *minterv1.CredentialsRequest) (*minterv
 	return nil, fmt.Errorf("no providerSpec defined")
 }
 
-// sync handles both create and update idempotently.
 func (a *AWSActuator) sync(ctx context.Context, cr *minterv1.CredentialsRequest) error {
+
 	logger := a.getLogger(cr)
 	logger.Debug("running sync")
 
+	// Should we update anything
+	needsUpdate, err := a.needsUpdate(ctx, cr)
+	if err != nil {
+		logger.WithError(err).Error("error determining whether a credentials update is needed")
+		return &actuatoriface.ActuatorError{
+			ErrReason: minterv1.CredentialsProvisionFailure,
+			Message:   fmt.Sprintf("error determining whether a credentials update is needed: %v", err),
+		}
+	}
+
+	if !needsUpdate {
+		return nil
+	}
+
+	cloudCredsSecret, err := a.getCloudCredentialsSecret(ctx, logger)
+	if err != nil {
+		logger.WithError(err).Error("issue with cloud credentials secret")
+		return err
+	}
+
+	if cloudCredsSecret.Annotations[secretannotator.AnnotationKey] == secretannotator.InsufficientAnnotation {
+		msg := "cloud credentials insufficient to satisfy credentials request"
+		logger.Error(msg)
+		return &actuatoriface.ActuatorError{
+			ErrReason: minterv1.InsufficientCloudCredentials,
+			Message:   msg,
+		}
+	}
+
+	if cloudCredsSecret.Annotations[secretannotator.AnnotationKey] == secretannotator.PassthroughAnnotation {
+		logger.Debugf("provisioning with passthrough")
+		err := a.syncPassthrough(ctx, cr, cloudCredsSecret, logger)
+		if err != nil {
+			return err
+		}
```

Comment:
I assume because it's just a secret copy, we safely assume that a failure in passthrough mode can be safely retried and doesn't need a condition?

Reply:
the syncPassthrough() call will create/return the ActuatorError, so we will end up setting the CredentialsProvisionFailure condition